models/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,130 @@
+package elasticsearch
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"biilabs.com/mam-data-explorer/models/config"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+)
+
+type DeviceLog struct {
+	Timestamp int64  `json:"timestamp"`
+	Message   string `json:"message"`
+	DeviceID  string `json:"deviceID"`
+}
+
+var (
+	ElasticClient *elasticsearch.Client
+)
+
+func init() {
+	cfg := elasticsearch.Config{
+		Addresses: []string{
+			config.ConfigHandler.GetString("ELASTICSEARCH_HOST"),
+		},
+	}
+
+	client, err := elasticsearch.NewClient(cfg)
+	if err != nil {
+		log.Fatalf("Error creating the client: %s", err)
+	}
+	ElasticClient = client
+}
+
+func InsertDeviceLog(deviceID string, message string) bool {
+	CurrentTime := time.Now().Unix()
+
+	deviceLog := DeviceLog{
+		Timestamp: CurrentTime,
+		Message:   message,
+		DeviceID:  deviceID,
+	}
+
+	// Marshal the struct to JSON and check for errors
+	jsonStr, err := json.Marshal(deviceLog)
+	if err != nil {
+		log.Println("json.Marshal ERROR:", err)
+		return false
+	}
+
+	documentID := strconv.FormatInt(CurrentTime, 10)
+	// Set up the request object.
+	req := esapi.IndexRequest{
+		Index:      deviceID,
+		DocumentID: documentID,
+		Body:       strings.NewReader(string(jsonStr)),
+		Refresh:    "true",
+	}
+
+	// Return an API response object from request
+	res, err := req.Do(context.Background(), ElasticClient)
+	if err != nil {
+		log.Fatalf("IndexRequest ERROR: %s", err)
+		return false
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		log.Printf("%s ERROR indexing document ID=%s", res.Status(), documentID)
+		return false
+	}
+
+	return true
+}
+
+func QueryDeviceLog(deviceID string) []DeviceLog {
+
+	type envelopeResponse struct {
+		Took int
+		Hits struct {
+			Total struct {
+				Value int
+			}
+			Hits []struct {
+				ID         string          `json:"_id"`
+				Source     json.RawMessage `json:"_source"`
```

Comment:
May I ask why do you need the underline?

Reply:
This is the json key defined by the elasticsearch response.
See also: https://github.com/elastic/go-elasticsearch